### PR TITLE
RELATED: RAIL-1531 add ExtendedDateFilter type guards

### PR DESCRIPTION
**Pull request by no23reason**

### CODE REVIEW

**Comment by yenkins** [comment]
Currently destined for [release R183](https://confluence.intgdc.com/display/Dev/R183).
This release is **zero downtime** release.
Code drop: 2019-06-01 00:00.

**Comment by yenkins** [comment]
Story [RAIL-1531](http://jira.intgdc.com/browse/RAIL-1531) Loading Date Filter Configuration from Project Metadata
	Status: Hacking
	Priority: Major
	Assignee: Rail Guild User

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/19/ : SUCCESS in 39s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/89/ : SUCCESS in 1m 49s

File: src/ExtendedDateFilters.ts [hunk 1]
```diff
@@ -77,20 +77,41 @@ export namespace ExtendedDateFilters {
         | IAbsoluteDateFilterForm
         | IAbsoluteDateFilterPreset;
 
-    export const isAbsoluteDateFilterOption = (option: DateFilterOption): option is AbsoluteDateFilterOption =>
+    export const isAllTimeDateFilter = (option: DateFilterOption): option is IAllTimeDateFilter =>
```
Comment (tomassatinsky) [inline src/ExtendedDateFilters.ts]: Where is this used?

Comment (no23reason) [reply src/ExtendedDateFilters.ts]: In KD, for example [here](https://github.com/gooddata/gdc-dashboards/blob/develop/app/services/ExtendedDateFilterService.ts#L219)

Comment (tomassatinsky) [reply src/ExtendedDateFilters.ts]: Makes sense

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-gate/10/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/90/ : SUCCESS in 1m 54s

**Pull request merged**